src/main/java/uk/gov/justice/hmpps/prison/service/MovementsService.java

```diff
@@ -314,6 +331,74 @@ private final String checkTransferParameters(final List<String> agencyIds, final
         return null;
     }
 
+    @Transactional
+    public OffenderMovement createExternalMovement(@NotNull final Long bookingId, final CreateExternalMovement createExternalMovement) {
+        final var offenderBooking = offenderBookingRepository.findById(bookingId)
+            .orElseThrow(EntityNotFoundException.withMessage("booking not found using %s", bookingId));
+
+        final var movementType = movementTypeRepository.findById(MovementType.pk(createExternalMovement.getMovementType()))
+            .orElseThrow(EntityNotFoundException.withMessage("movementType not found using: %s", createExternalMovement.getMovementType()));
+
+        final var movementReason = movementReasonRepository.findById(MovementReason.pk(createExternalMovement.getMovementReason()))
+            .orElseThrow(EntityNotFoundException.withMessage("movementReason not found using: %s", createExternalMovement.getMovementReason()));
+
+        final var movementReasons =
+            movementTypeAndReasonRespository.findMovementTypeAndReasonByTypeIs(createExternalMovement.getMovementType());
+
+        if (movementReasons.stream().noneMatch(r -> r.getReasonCode().equals(createExternalMovement.getMovementReason())))
+            throw new EntityNotFoundException("Invalid movement reason for supplied movement type");
+
+        if (offenderBooking.isActive())
+            throw new IllegalStateException("You can only create an external movement for inactive offenders");
+
+        final var fromAgency = agencyLocationRepository.findById(createExternalMovement.getFromAgencyId())
+            .orElseThrow(EntityNotFoundException.withMessage("fromAgency not found using: %s", createExternalMovement.getFromAgencyId()));
+
+        final var toAgency = agencyLocationRepository.findById(createExternalMovement.getToAgencyId())
+            .orElseThrow(EntityNotFoundException.withMessage("toAgency not found using: %s", createExternalMovement.getToAgencyId()));
+
+        final var externalMovement = ExternalMovement
+            .builder()
+            .offenderBooking(offenderBooking)
+            .movementSequence(0L)
+            .movementDate(createExternalMovement.getMovementTime().toLocalDate())
+            .movementTime(createExternalMovement.getMovementTime())
+            .fromAgency(fromAgency)
+            .toAgency(toAgency)
+            .movementDirection(createExternalMovement.getDirectionCode())
+            .movementType(movementType)
+            .movementReason(movementReason)
+            .activeFlag(ActiveFlag.Y)
+            .build();
+
+        offenderBooking.addExternalMovement(externalMovement);
```

Comment:
You need to check whether all previous movements need to be deactivated as well. If so, you also need to consider that the order of execution matters - see PrisonerReleaseAndTransferService.deactivatePreviousMovements

Reply:
I don't fully understand why we need to but I'm going to have a catch up with @Mjwillis once he's back to make sure he's happy with the approach overall.